frontend/src/components/ShoppingList.tsx

```diff
@@ -153,20 +157,31 @@ class ShoppingListList extends React.Component<IShoppingListContainerProps> {
           Copy
         </Button>
         <section ref={this.shoppingList}>
-          {items
-            .sort((x, y) => ingredientByNameAlphabetical(x[0], y[0]))
-            .map(([name, quantities], i) => {
-              if (quantities == null) {
-                return null
-              }
-              return (
-                <ShoppingListItem
-                  key={name}
-                  item={[name, quantities]}
-                  isFirst={i === 0}
-                />
-              )
-            })}
+          {groups.map(([groupName, values], groupIndex) => {
+            values.sort((x, y) => ingredientByNameAlphabetical(x[0], y[0]))
+            return (
+              <div key={groupName}>
+                {groupIndex > 0 && (
+                  // ensure copying the list has a new line between categories
+                  <section style={{ maxHeight: "0.5rem" }}>
```

Comment:
a little hacky 🤷🏻

Reply:
could use CSS if you wanted to avoid JS, but this seems like a nice JS solution